ports/stm/supervisor/port.c

```diff
@@ -290,13 +290,22 @@ condition and generating hardware reset or using Go command to execute user code
     HAL_RCC_DeInit();
     HAL_DeInit();
 
-    // disable all interupts
-    __disable_irq();
+    // Disable all pending interrupts using NVIC
+    for (uint8_t i = 0; i < (sizeof(NVIC->ICER) / sizeof(NVIC->ICER[0])); ++i) {
+        NVIC->ICER[i] = 0xFFFFFFFF;
+    }
 
-    // Clear all pending interrupts
-    for (uint8_t i = 0; i < (sizeof(NVIC->ICPR) / NVIC->ICPR[0]); ++i) {
```

Comment:
I looked at this many times before I spotted the problem.

In CircuitPython, you can use `MP_ARRAY_SIZE(arr)` (in py/misc.h) to compute the size of the array. Note that if the expression `arr` is actually a pointer instead, it will NOT be detected as an error. (the implementation is just `sizeof(arr) / sizeof(arr[0])`, more or less)

Reply:
@jepler did the changes. Testing now.